integration-tests/return-response/src/test/java/io/quarkiverse/openapi/generator/it/ReturnResponseTest.java

```diff
@@ -14,14 +16,26 @@
 class ReturnResponseTest {
 
     @Test
-    void testModel() throws NoSuchMethodException {
-        assertThat(ModelApi.class.getMethod("hello").getReturnType())
+    void testReturnResponseFalseString() throws NoSuchMethodException {
+        assertThat(ReturnResponseFalseStringApi.class.getMethod("hello").getReturnType())
                 .isEqualTo(String.class);
     }
 
     @Test
-    void testResponse() throws NoSuchMethodException {
-        assertThat(ResponseApi.class.getMethod("hello").getReturnType())
+    void testReturnResponseTrueString() throws NoSuchMethodException {
+        assertThat(ReturnResponseTrueStringApi.class.getMethod("hello").getReturnType())
+                .isEqualTo(Response.class);
+    }
+
+    @Test
+    void testReturnResponseFalseVoid() throws NoSuchMethodException {
+        assertThat(ReturnResponseFalseVoidApi.class.getMethod("hello").getReturnType())
+                .isEqualTo(Void.TYPE);
```

Comment:
```suggestion
                .isEqualTo(Response.class);
```

See comment https://github.com/quarkiverse/quarkus-openapi-generator/issues/424#issuecomment-1656747394